src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxnidqinterface.py

```diff
@@ -0,0 +1,144 @@
+"""Authors: Cody Baker."""
+import json
+from pathlib import Path
+from typing import List
+
+import numpy as np
+from pynwb.ecephys import ElectricalSeries
+
+from .spikeglxdatainterface import SpikeGLXRecordingInterface
+from .spikeglx_utils import get_session_start_time
+from ....tools.signal_processing import parse_rising_and_falling_frames_from_ttl
+from ....utils import get_schema_from_method_signature, get_schema_from_hdmf_class, FilePathType, dict_deep_update
+
+
+class SpikeGLXNIDQInterface(SpikeGLXRecordingInterface):
+    """Primary data interface class for converting the high-pass (ap) SpikeGLX format."""
+
+    @classmethod
+    def get_source_schema(cls):
+        source_schema = get_schema_from_method_signature(class_method=cls.__init__, exclude=["x_pitch", "y_pitch"])
+        source_schema["properties"]["file_path"]["description"] = "Path to SpikeGLX .nidq file."
+        return source_schema
+
+    def __init__(
+        self,
+        file_path: FilePathType,
+        stub_test: bool = False,
+        spikeextractors_backend: bool = False,
+        verbose: bool = True,
+    ):
+        """
+        Read channel data from the NIDQ board for the SpikeGLX recording.
+
+        Useful for synchronizing multiple data streams into the common time basis of the SpikeGLX system.
+
+        Parameters
+        ----------
+        file_path: FilePathType
+            Path to .nidq.bin file.
+        stub_test: bool
+            Whether to shorten file for testing purposes. Default: False.
+        verbose: bool
+            Whether to output verbose text. Default: True.
+        """
+        self.stream_id = "nidq"
+
+        folder_path = Path(file_path).parent
+        super().__init__(folder_path=folder_path, stream_id=self.stream_id, verbose=verbose)
+        self.source_data.update(file_path=str(file_path))
+
+        self.meta = self.recording_extractor.neo_reader.signals_info_dict[(0, self.stream_id)]["meta"]
+
+    def get_metadata_schema(self):
+        metadata_schema = super().get_metadata_schema()
+        metadata_schema["properties"]["Ecephys"]["properties"].update(
+            ElectricalSeriesRaw=get_schema_from_hdmf_class(ElectricalSeries)
```

Comment:
This goes back to our conversation about what an ElectricalSeries should be

Reply:
Yep, it's the least amount of work right now to just consider it another `ElectricalSeries`.

Would be a little extra work to setup a separate table from the main `nwbfile.electrodes` plus a `DynamicTableRegion` pointing to the custom one.

Otherwise, would lose device/group metadata information by writing directly to a `TimeSeries` instead.